test/TestUtil.js

```diff
@@ -30,6 +29,30 @@ exports.promiseLater = function (time, func) {
     }));
 };
 
+/**
+ * Returns rejection reason if rejected, otherwise returns a dummy error.
+ */
+exports.getRejectionReasonOrDummy = async function (object, asyncMethodName, ...params) {
```

Comment:
I think we could call it just `getRejectionReasonOrThrow` to make it easier to understand